integrationtests/test_all_examples.py

```diff
@@ -0,0 +1,312 @@
+#  Copyright (c) 2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+
+""" Tests for demo and tutorial examples.
+"""
+
+import contextlib
+import io
+import os
+import sys
+import traceback
+import unittest
+from unittest import mock
+
+from traits.api import HasTraits
+
+from traitsui.tests._tools import (
+    is_current_backend_wx,
+    is_current_backend_qt4,
+    process_cascade_events,
+    skip_if_null,
+    store_exceptions_on_all_threads,
+)
+
+# This test file is not distributed nor is it in a package.
+HERE = os.path.dirname(__file__)
+
+
+class ExampleSearcher:
+    """ This object collects and reports example files to be tested."""
+
+    def __init__(self, source_dirs):
+        """
+        Parameters
+        ----------
+        source_dirs : list of str
+            List of directory paths from which Python files will be collected.
+        """
+        self.source_dirs = source_dirs
+        self.files_may_be_skipped = {}
+
+    def skip_file_if(self, condition, reason, filepath):
+        """ Mark a file to be skipped for a given condition.
+
+        Parameters
+        ----------
+        condition: callable() -> bool
+            The condition for skipping a file.
+        reason : str
+            Reason for skipping the file.
+        filepath : str
+            Path of the file which may be skipped from tests.
+        """
+        filepath = os.path.abspath(filepath)
```

Comment:
If I did not have this line, the test will fail to skip the files appropriately, e.g. if the `filepath` given to `skip_file_if` is not an absolute path but `is_skipped` is called with an absolute path.

Reply:
Yes, I know. But if `EXAMPLES` was saved as an absolute path, then all paths given to `skip_file_if` would also be absolute paths without this line.

Reply:
Anyways, now that the code has been restructured, the registration via `skip_file_if` doesn't stand out as much, so it can stay as is, except for maybe changing the order of arguments (see comment above).

Reply:
I see.